src/abstractdataframe/reshape.jl

```diff
@@ -55,14 +57,19 @@ function stack(df::AbstractDataFrame, measure_vars::AbstractVector{<:Integer},
                id_vars::AbstractVector{<:Integer}; variable_name::Symbol=:variable,
                value_name::Symbol=:value, view::Bool=false,
                variable_eltype::Type=CategoricalString)
+    if stack_didnotwarn
```

Comment:
I'm not sure it's a good idea to print this. There's no way to turn this off so it will annoying everybody. And if you relied on the columns order, you'll find out the problem soon enough.

The only way to deprecate this cleanly would be to introduce a keyword argument, but that would be quite annoying.

Reply:
The point is that I did not do a normal deprecation exactly because I do not want the message to be annoying. It will be printed only once per Julia session (as opposed a normal depwarn that would be printed every time).

The deprecation would only be in 0.21, in 1.0 release it will be gone.

@oxinabox, @pdeffebach  - can you please comment on it from a "user's voice" perspective. Thank you!

Reply:
> as opposed a normal depwarn that would be printed every time).

is that still not fixed?

Rather than using a global  you can do `@warn msg maxlog=1`
to prevent warning showing up more than once

---

I never use `stack`, only `unstack`, so i have no stake in this game.

Warnings are, in general are incredibly annoying.
I would rather a well-documented and breaking change than most warnings.

Reply:
> `@warn msg maxlog=1`

Thank you - one learns something all the time (I knew there should be a solution for this :smile:). If we end up printing a deprecation warning I will change the implementation to use your tip.